2020-04-introducing-mdjs-interactive-demos-everywhere/README.md

```diff
@@ -0,0 +1,363 @@
+---
+title: Introducing mdjs - interactive demos everywhere
+published: false
+description: Combine markdown and javascript to create the ultimate documentation to allow interactive demos everywhere.
+tags: markdown, javascript, webcomponents, demos
+cover_image: https://raw.githubusercontent.com/open-wc/blog-posts/feat/mdjs/2020-04-introducing-mdjs-interactive-demos-everywhere/images/aaron-burden-TNlHf4m4gpI-unsplash.jpg
+---
+
+Every shared code needs had written documentation to show what it can be used for and what the idea behind is.
+Users should at least be able to get a high level understanding of what they are using, what they're using it for, and why.
+
+On the web, we have many many different ways of writing documentation.
+However, one thing almost all of them have in common is that they rely on [Markdown](https://en.wikipedia.org/wiki/Markdown) or some kind of variation of it.
+
+And it's no surprise, because Markdown is supported practically everywhere (vscode, atom, github, gitlab, dev.to, npmjs, ...)
+
+## For tools that do not run in the browser
+
+In that case, you will mostly share code snippets that people will need to run in their own projects so traditional static site generators like [Docusaurus](https://docusaurus.io/), [VuePress](https://vuepress.vuejs.org/), [Gatsby](https://www.gatsbyjs.org/), ... work great. All of them have full support for Markdown and allow you to easily create beautiful documentation pages with code snippets/highlighting, and more.
+
+And frankly, if that is your use case almost everything you need should be possible with those tools as long as you feel comfortable with the ecosystem/framework.
+
+## For (visual) components that do run in the browser
+
+In this context, users probably do expect a live demo to see all the different options of your component in action. So pure Markdown is usually not enough as we now want to actually execute code and "insert" our working component in our documentation somehow. This would require specialized handling for each framework.
+
+### Vue
+
+For Vue you can use for example VuePress which auto registers all vue components in a certain folder and then you can use as normal html tags as Markdown allows html
```

Comment:
```suggestion
For Vue, as an example, you can use VuePress which auto registers all Vue components in a certain folder and then you can use as normal html tags since Markdown supports html
```